nexus/src/app/sagas/instance_delete.rs

```diff
@@ -0,0 +1,106 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
```

Comment:
Is there a reason this is a saga and not just a transaction that does all three of these things?   Is it just the lack of good support for non-interactive transactions?

Reply:
Yeah, basically. Any saga steps that purely talk to CRDB could theoretically be combined, but my main goals were:

1. Make sure we have crash-consistency (before: we weren't durable, now, we are)
2. Avoid interactive transactions. They're slow, and even worse, their contention has resulted in CRDB errors like https://www.cockroachlabs.com/docs/stable/transaction-retry-error-reference.html#retry_write_too_old . Those errors are a bummer, especially for high-volume operations, because they consume compute time and *don't necessarily make forward progress*. 

I'm very open to smushing these DB-only steps together, but putting them in a saga felt like a reasonable intermediate step.

(Also, when we add dendrite integration, I think we're going to have some non-DB-only steps here *anyway*, so having this operation be a saga is something we will probably want)

Reply:
I was also doing some reading on https://www.cockroachlabs.com/docs/stable/transactions.html#batched-statements , which seems great, but has the caveat: the ability for CRDB to perform an automatic retry without another network round-trip has some caveats:
- The size of results must be less than 16 KiB, or automatic retry is disabled
- If any part of the transaction relies on conditional logic, or automatic retry is disabled

I still think it's worth it to pursue this path, but it felt like an "optimization" above multiple saga steps. In contrast, the following felt like correctness issues:
- Lack of durability across crashes
- Throwing errors unnecessarily back to the client because of database contention